sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -321,7 +321,8 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
     assert(spark.sharedState.cacheManager.isEmpty)
   }
 
-  test("Clear accumulators when uncacheTable to prevent memory leaking") {
+  // This test would be flaky.
+  ignore("Ensure accumulators to be cleared after GC when uncacheTable") {
```

Comment:
This is the only risky part of this PR, I'll think about how to deterministically test it